plugins/GCodeProfileReader/GCodeProfileReader.py

```diff
@@ -51,31 +55,36 @@ def read(self, file_name):
         # Loading all settings from the file.
         # They are all at the end, but Python has no reverse seek any more since Python3.
         # TODO: Consider moving settings to the start?
-        serialised = ""  # Will be filled with the serialised profile.
+        serialized = ""  # Will be filled with the serialized profile.
         try:
             with open(file_name) as f:
                 for line in f:
                     if line.startswith(prefix):
                         # Remove the prefix and the newline from the line and add it to the rest.
-                        serialised += line[prefix_length : -1]
+                        serialized += line[prefix_length : -1]
         except IOError as e:
             Logger.log("e", "Unable to open file %s for reading: %s", file_name, str(e))
             return None
-
-        # Un-escape the serialised profile.
+        
+        # Un-escape the serialized profile.
         pattern = re.compile("|".join(GCodeProfileReader.escape_characters.keys()))
 
         # Perform the replacement with a regular expression.
-        serialised = pattern.sub(lambda m: GCodeProfileReader.escape_characters[re.escape(m.group(0))], serialised)
+        serialized = pattern.sub(lambda m: GCodeProfileReader.escape_characters[re.escape(m.group(0))], serialized)
+        Logger.log("i", "Serialized the following from %s: %s" %(file_name, repr(serialized)))
 
-        # Apply the changes to the current profile.
-        profile = Profile(machine_manager = Application.getInstance().getMachineManager(), read_only = False)
+        # Create an empty profile - the id will be changed later
+        profile = InstanceContainer("G-code-imported-profile")
```

Comment:
The name "G-code-imported-profile" is subsequently replaced, isn't it? Then why introduce this hardcoded name?
"" works too.